system/lib/pthread/library_pthread.c

```diff
@@ -745,16 +745,20 @@ EMSCRIPTEN_KEEPALIVE double emscripten_run_in_main_runtime_thread_js(int index,
   c->calleeDelete = 1-sync;
   c->functionEnum = EM_PROXIED_JS_FUNCTION;
   c->functionPtr = (void*)index;
-  // We write out the JS doubles into args[], which must be of appropriate size - JS will assume that.
+  // The types are only known at runtime in these calls, so we store values that
+  // must be able to contain any valid JS value, including a 64-bit BigInt if
+  // BigInt support is enabled.
   assert(sizeof(em_variant_val) == sizeof(double));
```

Comment:
Does it still make sense to have this first assert?     Also shouldn't these but `>=`?   It seems like if you include both of these you are needlessly asserting that `double` and `int64_t` are the same size which is not really important to this code is it?

I assume this assert its trying to say "Make sure that the size of the variant is at least is a big as the biggest thing I need to store in it?

I'm actually not sure you need these assertions at all since you are not doing any kind of casting.. the type system should take care of these invariants, no?

If you decide to keep them how about `static_assert()`?

Reply:
Yes, basically these are just asserting that the slot size is big enough for the biggest JS things.

Maybe these have little value, as there is no real chance a double would not be 64 bits... I'll remove them.